lib/Solver/Z3Solver.cpp

```diff
@@ -210,7 +210,7 @@ char *Z3SolverImpl::getConstraintLog(const Query &query) {
 }
 
 bool Z3SolverImpl::computeTruth(const Query &query, bool &isValid) {
-  bool hasSolution;
+  bool hasSolution = false; // to remove compiler warning
```

Comment:
I don't think this comment belongs here.

Reply:
Maybe, but without it it's confusing as to why this is needed, or why this is not `= false` (I could have also chosen `true`).  In other words, the code before was nicer, but I prefer to get rid of compiler warnings.  But if you feel strongly against it, I can remove the comment.

Reply:
Ah good point. Could we just remove `hasSolution` and then have `isValid = !isValid`? I guess that's a bit confusing to read. I don't feel strongly about this.